coderd/database/queries/templateversions.sql

```diff
@@ -110,3 +110,17 @@ SET
 	updated_at = $3
 WHERE
 	job_id = $1;
+
+-- name: GetPreviousTemplateVersion :one
+SELECT
+	*
+FROM
+	template_versions
+WHERE
+	created_at < (
+		SELECT created_at
+		FROM template_versions as tv
+		WHERE tv.organization_id = $1 AND tv.name = $2 AND tv.template_id = $3
```

Comment:
Why do we need both `template_id` and `name` (I would think `template_id` would suffice)?

Also, don't we need ORDER BY And LIMIT here to ensure we're only comparing against one row?

Reply:
> Why do we need both template_id and name (I would think template_id would suffice)?
Because two different templates can have versions with the same name. Eg. Templates `docker` and `gcloud` can have versions `latest`, `alpha`, `beta` etc.

> Also, don't we need ORDER BY And LIMIT here to ensure we're only comparing against one row?
Not necessary since the WHERE is already doing that. Not sure if we should create some sort of index for that to ensure this constraint.